### DISCUSSION TITLE
How to mark the blog as home page, when I have a documentation website too

### DISCUSSION THREAD

**Discussion by janban-outlook** [squidfunk/mkdocs-material#6721] [Q&A] [answered]
I want my blog to be the main entry point to my website. In the mkdocs.yml file, I structured the navigation section as follows:
```
nav:
  - Blog: blog/index.md
  - About me: index.md
  - other page: page2.md
  - yet another page: page3.md
  - Contact: contact.md
 ```

This successfully moves the Blog to the first position in the menu. However, upon opening the site, the About page is displayed by default, which is expected since it is the index page. To address this, I renamed it to about.md. But this change caused an issue as the website then lacked an index page.

The only solution I can think of is to create an index page that redirects to the blog, like this:
```
---
template: redirect.html
location: blog/
---
```

Is this the proper approach, or am I overlooking a setting that I can add to the mkdocs.yml file?

**Comment by kamilkrzyskow** [comment]
Hello @janvanveldhuizen,
MkDocs generates static pages, simple HTML files, it doesn't affect how the files are being accessed. 
If you have control over your server implementation you could redirect the request of `/index.html` to `/blog/index.html`, either modifying `.htaccess` or some nginx configuration etc. I don't think that GitHub Pages allow for such changes 🤔 

Apart of that there is the solution like you described, which is - after the `/index.html` is accessed and loaded, redirect the user to `/blog/index.html`.

As a side note, perhaps this is a valid change request for the Blog plugin to support creating the blog index page as the root of the website, but keep the posts under the `/blog/` subpath, as I believe the current setup separating the blog from simple docs was made to avoid any path collisions 🤔

**Reply by janban-outlook** [reply]
So far, my own solution works perfectly. I think I keep it like that.

**Comment by squidfunk** [comment, marked as answer]
You can follow the standalone guide as outlined [in this blog post](https://squidfunk.github.io/mkdocs-material/blog/2022/09/12/blog-support-just-landed/), and then just change the [`post_url_format`](https://squidfunk.github.io/mkdocs-material/plugins/blog/#config.post_url_format) to:

``` yaml
plugins:
  - blog:
      post_url_format: "blog/{date}/{slug}"
```

If you wish to host the archive and categories under `blog` as well, adjust [`archive_url_format`](https://squidfunk.github.io/mkdocs-material/plugins/blog/#config.archive_url_format) and [`categories_url_format`](https://squidfunk.github.io/mkdocs-material/plugins/blog/#config.categories_url_format). The blog plugin is extremely flexible and allows for a lot of freedom when it comes to URL generation. If something is missing in terms of freedom, we can consider adding it.

**Reply by janban-outlook** [reply]
I do not have a standalone blog. I have the other pages next to it. But I'll see how far I can come using this approach.
But the redirect solution works perfectly anyway.

**Reply by squidfunk** [reply]
No problem. Standalone blog means it lives at the top, and not in a subdirectory of your documentation. You can have as many pages next to it as you like. However, if you found another solution, no need to change it.

I just posted it for completeness.
